test/com/google/enterprise/adaptor/opentext/OpentextAdaptorTest.java

```diff
@@ -3053,6 +3556,137 @@ public GroupPrincipal pushGroupDefinitions(
     assertEquals(DocIdPusher.FeedType.INCREMENTAL, pusher.latestFeedType);
   }
 
+  @Test
+  public void testGroupsNullSearchResults() throws InterruptedException {
+    class MemberServiceMockError extends MemberServiceMock {
+      @Override
+      public MemberSearchResults getSearchResults(PageHandle pageHandle) {
+        return null;
+      }
+    }
+    SoapFactoryMock soapFactory = new SoapFactoryMock();
+    soapFactory.memberServiceMock = new MemberServiceMockError();
+    soapFactory.memberServiceMock.addMember(
+        getMember(2000, "group 1", "Group"));
+    OpentextAdaptor adaptor = new OpentextAdaptor(soapFactory);
+    AdaptorContext context = ProxyAdaptorContext.getInstance();
+    Config config = initConfig(adaptor, context);
+    adaptor.init(context);
+    OpentextAdaptor.Groups groups =
+        adaptor.new Groups(soapFactory.newMemberService());
+    groups.addGroups();
+    assertEquals(0, groups.getGroupDefinitions().size());
```

Comment:
These should either include groups.getGroupDefinitions().toString(), or perhaps better, use full equality (if it's a list, set, or map).